server/app/app.py

```diff
@@ -30,13 +193,25 @@ def create_app(self):
         # Config
         SECRET_KEY = os.environ.get("CXG_SECRET_KEY", default="SparkleAndShine")
         self.app.config.update(SECRET_KEY=SECRET_KEY)
-        self.app.config.update(SCRIPTS=[])
 
-        resources = get_api_resources()
-        self.app.register_blueprint(webapp.bp)
-        self.app.register_blueprint(resources.blueprint)
-        self.app.add_url_rule("/", endpoint="index")
+        self.app.register_blueprint(webbp)
+
+        api_version = "/api/v0.2"
+        if data:
+            bp_api = Blueprint("api", __name__, url_prefix=api_version)
+            resources = get_api_resources(bp_api)
+            self.app.register_blueprint(resources.blueprint)
+            self.app.add_url_rule("/", "index", index)
+
+        if app_config.dataroot:
+            bp_api = Blueprint("api_dataset", __name__, url_prefix="/<dataset>" + api_version)
+            resources = get_api_resources(bp_api)
+            self.app.register_blueprint(resources.blueprint)
+            self.app.add_url_rule("/<dataset>/", 'dataset_index', dataset_index)
```

Comment:
Longer term, we should consider if we want to support accessing datasets from sub-directories inside the dataroot.  This route format doesn't allow it (all must be at the top level).  I think it is fine for this PR - just wanted to make a note.

Reply:
added  a note